tests/integration/states/test_file.py

```diff
@@ -844,6 +844,26 @@ def test_managed_local_source_does_not_exist(self):
             # Check that we identified a hash mismatch
             self.assertIn("does not exist", ret["comment"])
 
+    def test_test_managed_issue_55269(self):
+        """
+        Make sure that we exit gracefully in case a local source does not exist
+        when file.managed is run with test=1.
+        """
+        name = self.tmp_dir / "local_source_does_not_exist_testing"
+        self.addCleanup(salt.utils.files.safe_rm, str(name))
+        local_path = os.path.join(RUNTIME_VARS.BASE_FILES, "grail", "scene99")
+
+        for proto in ("file://", ""):
+            source = proto + local_path
+            log.debug("Trying source %s", source)
+            ret = self.run_state(
+                "file.managed", name=str(name), source=source, test=True
+            )
+            self.assertSaltFalseReturn(ret)
+            ret = ret[next(iter(ret))]
+            self.assertFalse(ret["changes"])
+            self.assertIn("does not exist".format(), ret["comment"])
```

Comment:
```suggestion
            self.assertIn("does not exist", ret["comment"])
```

Reply:
Sorry I didn't catch that. Fixed and rebased on current master.